NDTensors/ext/NDTensorsCUDAExt/typeparameteraccessors.jl

```diff
@@ -0,0 +1,20 @@
+# # `TypeParameterAccessors.jl` overloads.
+using NDTensors.TypeParameterAccessors: TypeParameterAccessors, Position, parameter
+## TODO this seems like a `GPUArrays` generic function
+storagemode(t::Type{<:CuArray}) = parameter(t, 3)
+function TypeParameterAccessors.position(::Type{<:CuArray}, ::typeof(storagemode))
+  return Position(3)
+end
+
+function TypeParameterAccessors.default_parameter(::Type{<:CuArray}, ::typeof(eltype))
+  return Float64
+end
+TypeParameterAccessors.default_parameter(::Type{<:CuArray}, ::typeof(ndims)) = 1
+function TypeParameterAccessors.default_parameter(::Type{<:CuArray}, ::typeof(storagemode))
+  return Mem.DeviceBuffer
+end
+
+## TODO this seems like a `GPUArrays` generic function
+function set_storagemode(arraytype::Type{<:CuArray}, store)
+  return set_parameter(arraytype, storagemode, store)
+end
```

Comment:
```suggestion
function set_storagemode(arraytype::Type{<:CuArray}, param)
  return set_parameter(arraytype, storagemode, param)
end
```
Just a small style change, but I prefer to pick names that could be chosen consistently across many similar functions (i.e. make them more general instead of too particular to the function).

Reply:
Okay this makes sense. Before I finish this PR I will read through everything again and make sure all of these functions are consistent in the way you describe here. I have updated this instance now (I am working through your comments first before going through and reading the code again.)